python/social_bot/tasks.py

```diff
@@ -598,18 +652,27 @@ def task_specific_observation(self, agent):
                 pose = np.concatenate((pose, obj_pos), axis=0)
 
         agent_pose = np.array(agent.get_pose()).flatten()
-        if self._use_egocentric_states:
+        if self._use_full_states or self._use_egocentric_states:
```

Comment:
what is the different between _use_full_states  or not?  task_specific_observation is suposed to be full states of the world. If essentially they are similar, remove the part in else below:
```
            agent_vel = np.array(agent.get_velocities()).flatten()
            joints_states = agent.get_internal_states()
            obs = np.concatenate((pose, agent_pose, agent_vel, joints_states),
                                 axis=0)
```

Reply:
full_states so far excludes agent's internal_states, but thinking of it, maybe we should include agent's internal states into aux_achieved (similar as agent's speed).

Reply:
Internal states might be useless here for goal task. They are angles and velocity of two wheels. Pretty much another representation of position changes, velocity, yaw and yaw rate that are already in agent_vel. I think it's fine to remove this part.